src/main/java/org/broadinstitute/hellbender/utils/genotyper/AlleleLikelihoods.java

```diff
@@ -758,46 +759,28 @@ private void extendsLikelihoodArrays(final double initialLikelihood, final int s
         }
     }
 
-    // Append the new evidence reference into the structure per-sample.
-    private List<EVIDENCE> appendEvidence(final List<EVIDENCE> newSampleEvidence, final int sampleIndex) {
+    // Append the new evidence reference into the structure per-sample, returning the count of evidence actually added (duplicates are not added)
+    // NOTE: the evidence-to-index cache is updated in place and not invalidated via {@link #invalidateEvidenceToIndexCache(int)} because adding new evidence
+    // to the cache, as opposed to removing evidence, is just a matter of appending entries
+    private int appendEvidence(final List<EVIDENCE> newSampleEvidence, final int sampleIndex) {
 
         final List<EVIDENCE> sampleEvidence = evidenceBySampleIndex.get(sampleIndex);
         final Object2IntMap<EVIDENCE> sampleEvidenceIndex = evidenceIndexBySampleIndex(sampleIndex);
+        final int previousEvidenceCount = sampleEvidence.size();
 
-        // actually-added will have the list of evidence added at the end of this method.
-        // this won't include those that were already in the table.
-        // being optimistic we assume that there is no repeats in the input new evidence so we set it to
-        // the input list but if we found something we then start a new list.
-        List<EVIDENCE> actuallyAdded = newSampleEvidence;
-
-        int i, nextIndex = sampleEvidence.size();
-        final int stop = newSampleEvidence.size();
-        for (i = 0; i < stop; i++) {
-            final EVIDENCE newEvidence = newSampleEvidence.get(i);
-            final int previousValue = sampleEvidenceIndex.put(newEvidence, nextIndex);
-            if (previousValue == MISSING_INDEX) {
-                nextIndex++;
-                sampleEvidence.add(newEvidence);
-            } else {
-                actuallyAdded = new ArrayList<>(newSampleEvidence.subList(0, i));
-                i++; // skip the repeated element.
-                break;
-            }
-        }
-        // second for below only use if we encounter some evidence that is already in the table:
-        for (; i < stop; i++) {
-            final EVIDENCE newEvidence = newSampleEvidence.get(i);
+        int nextIndex = sampleEvidence.size();
+        for (final EVIDENCE newEvidence : newSampleEvidence) {
             final int previousValue = sampleEvidenceIndex.put(newEvidence, nextIndex);
             if (previousValue == MISSING_INDEX) {
                 nextIndex++;
                 sampleEvidence.add(newEvidence);
-                actuallyAdded.add(newEvidence);
             } else {
                 sampleEvidenceIndex.put(newEvidence, previousValue); // revert
             }
         }
+
         numberOfEvidences[sampleIndex] = sampleEvidence.size();
-        return actuallyAdded;
+        return sampleEvidence.size() - previousEvidenceCount;
```

Comment:
```return nextIndex - previousEvidenceCount;```

Reply:
True, but upon further thought I realized this method should be `void` because its return value is not really needed.